core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SPARQLProtocolSession.java

```diff
@@ -880,6 +866,31 @@ protected void getRDF(HttpUriRequest method, RDFHandler handler, boolean require
 		}
 	}
 
+	/**
+	 * Pass through response content directly to the supplied sink if possible.
+	 * 
+	 * @param response       the {@link HttpResponse} with the content.
+	 * @param responseFormat the format of the response.
+	 * @param sink           the {@link Sink} to pass the content through to.
+	 * @return {@code true} if the content was passed through, {@code false} otherwise.
+	 * @throws IOException
+	 */
+	private boolean passThrough(HttpResponse response, FileFormat responseFormat, Sink sink)
+			throws IOException {
+		if (responseFormat.equals(sink.getFileFormat())) {
+			InputStream in = response.getEntity().getContent();
+			if (sink instanceof CharSink) {
+				IOUtils.copy(in, ((CharSink) sink).getWriter(),
+						getResponseCharset(response).orElse(responseFormat.getCharset()));
```

Comment:
Maybe introduce a variable for the used Charset?

Reply:
My personal rule of thumb is not to introduce a variable unless you refer to it at least twice and/or the code becomes unreadable without it. YMMV but I think this is still in the "easy enough to read" category :)

Reply:
A programmer will always have a personal preference. 😉 I am fine with that, and agree that this case is in the category "easy enough to read".